### DISCUSSION TITLE
The Mux situation with sing-box and clash

### DISCUSSION THREAD

**Discussion by radiumatic** [XTLS/Xray-core#3635] [Q&A]
I've been trying to find a solution for the past few months and unfortunately I haven't been able to. sing-box and clash are objectively easier to use as clients because not only they support tun, they bundle multiple protocols. However, you can't use mux with a xray-core server and a sing-box or clash client, which can become really annoying in some situations. 

I wanted to know how the community deals with this. What do you use?

P.S: I've been curious for some time now. What is the general preference in the Chinese community for anti-censorship tools? In Iran, mostly xray-core and sing-box are used as servers and clients, along with some modified cores and clients. (Eg, MahsaNG and Hiddify Next)

**Comment by Artoria2e5** [comment]
You are right that there's a divide in how Mux is done. It's pretty regrettable once you realize Mux is quite important for running UDP at all; or in the case of Shadowsocks, for not running udp (DNS, bittorrent μTP -- well, I do whatever I do with my own VPS!) over udp, which in my experience is quite obvious to the censor and quickly leads to banning.

> I wanted to know how the community deals with this. 

The community simply does not deal with this. Clients default to using no Mux. Which is, well, bad for getting UDP to work.

> What is the general preference in the Chinese community for anti-censorship tools? 

We're split, mostly evenly, two ways on the server site between sing-box and Xray, though there might be some more preference for Xray because there's a nice web panel (x-ui). On the client side, most people use either Clash Verge Rev (with Mihomo [formerly Clash-Meta] core) GUI, or Windows V2RayN GUI (which typically uses Xray, but also supports sing-box -- and indeed almost always uses sing-box locally for TUN). Sing-box loses out by not having a GUI except on mobile phones and macOS (where it works quite well and has like half the market share).

**Reply by radiumatic** [reply]
That's really interesting. There's also NekoRay (and the android counterpart NekoBox) and S-UI (sing-box web UI) if you ever want to check them out.

Well, Hiddify Next exists too (Linux, Windows, Mac, iOS, Android) but it's been quite the bloat lately. 

Do you mind trying some patched clients built for Iran? The upstream projects reject the changes (TLS fragmentation, WireGuard noise), saying they won't work in China. It would be neat to have some proof that they work there.
